src/interfaces/HposInterface.js

```diff
@@ -18,53 +21,72 @@ export const HPOS_API_URL = HPOS_PORT
   ? `http://localhost:${HPOS_PORT}`
   : (window.location.protocol + '//' + window.location.host)
 
-function hposCall ({ pathPrefix, method = 'get', path, headers: userHeaders = {} }) {
-  return async params => {
-    const fullUrl = HPOS_API_URL + pathPrefix + path
+async function hposCall ({ pathPrefix, method = 'get', path, headers: userHeaders = {}, params }) {
+  const fullUrl = HPOS_API_URL + pathPrefix + path
 
-    const signature = await signRequest(method, fullUrl, params)
-    const headers = {
-      ...axiosConfig.headers,
-      ...userHeaders,
-      'X-Hpos-Admin-Signature': signature
-    }
+  const authToken = localStorage.getItem('authToken')
+  // If localStorage.getItem('authToken') == null then hposCall should return error 401
+  if (localStorage.getItem('authToken') == null) {
+    return Promise.reject({response: { status: 401}})
+  }
 
-    let data
-
-    switch (method) {
-      case 'get':
-        ({ data } = await axios.get(fullUrl, { params, headers }))
-        return data
-      case 'post':
-        ({ data } = await axios.post(fullUrl, params, { headers }))
-        return data
-      case 'put':
-        ({ data } = await axios.put(fullUrl, params, { headers }))
-        return data
-      case 'delete':
-        ({ data } = await axios.delete(fullUrl, { params, headers }))
-        return data
-      default:
-        throw new Error(`No case in hposCall for ${method} method`)
-    }
+  const headers = {
+    ...axiosConfig.headers,
+    ...userHeaders,
+    'X-Hpos-Auth-Token': authToken
   }
-}
 
-export function hposAdminCall (args) {
-  return hposCall({
-    ...args,
-    pathPrefix: '/api/v1'
-  })
+  let data
+
+  switch (method) {
+    case 'get':
+      ({ data } = await axios.get(fullUrl, { params, headers }))
+      return data
+    case 'post':
+      ({ data } = await axios.post(fullUrl, params, { headers }))
+      return data
+    case 'put':
+      ({ data } = await axios.put(fullUrl, params, { headers }))
+      return data
+    case 'delete':
+      ({ data } = await axios.delete(fullUrl, { params, headers }))
+      return data
+    default:
+      throw new Error(`No case in hposCall for ${method} method`)
```

Comment:
I am wondering if we should just return a `404` if the method name is not found

Reply:
True! But if you don't mind I am gonna leave it as is in this PR